code/game/objects/items/devices/powersink.dm

```diff
@@ -120,13 +121,62 @@
 /obj/item/device/powersink/process()
 	drained_this_tick = 0
 	power_drained -= min(dissipation_rate, power_drained)
-	if(power_drained > max_power * 0.95)
-		playsound(src, 'sound/effects/screech.ogg', 100, 1, 1)
-	if(power_drained >= max_power)
-		explosion(src.loc, 3,6,9,12)
-		qdel(src)
-		return
+
 	if(attached && attached.powernet)
 		PN = attached.powernet
 	else
 		PN = null
+
+	if(power_drained > max_power * 0.98)	// Lower the screeching period. It was pretty long during testing.
+		playsound(src, 'sound/effects/screech.ogg', 100, 1, 1)
+
+	if(power_drained >= max_power)
+		handle_overload()
+		qdel(src)
+		return
+
+/obj/item/device/powersink/proc/handle_overload()
+	if (QDELETED(src))
+		return
+
+	// No attached node, or no powernet.
+	if (!PN)
+		explosion(src.loc, 0, 1, 3, 6)
+		return
+
+	// Propagate the power surge through the powernet nodes.
+	for (var/A in PN.nodes)
+		if (!A || A == src)
+			continue
+
+		var/dist = get_dist(src, A)
+
+		if (dist < 1)
+			dist = 1	// For later calculations.
+		else if (dist > 24)
+			continue	// Out of range.
+
+		// Map it to a range of [3, 1] for severity.
+		dist = round(MAP(dist, 1, 28, 3, 1))
+
+		// Check for terminals and affect their master nodes. Also add a special
+		// case for APCs whereby their lights are popped or flicked.
+		if (istype(A, /obj/machinery/power/terminal))
+			var/obj/machinery/power/terminal/T = A
+			if (istype(T.master, /obj/machinery/power/apc))
+				var/obj/machinery/power/apc/AP = T.master
+				if (dist > 1)
+					AP.overload_lighting(100, TRUE)
+				else
+					AP.flicker_all()

```

Comment:
How many timers do you see when this is triggered? I recall this proc creating ~100 timers for just a single area.